src/client/dfuse/dfuse_core.c

```diff
@@ -1194,26 +1252,42 @@ dfuse_fs_start(struct dfuse_projection_info *fs_handle, struct dfuse_cont *dfs)
 	if (rc != -DER_SUCCESS)
 		D_GOTO(err_ie_remove, rc);
 
-	rc = d_slab_register(&fs_handle->dpi_slab, &read_slab, &fs_handle->dpi_read_slab);
-	if (rc != -DER_SUCCESS)
-		D_GOTO(err_slab, rc);
+	for (i = 0; i < fs_handle->dpi_eqt_count; i++) {
+		struct dfuse_eq *eqt = &fs_handle->dpi_eqt[i];
 
-	rc = d_slab_register(&fs_handle->dpi_slab, &write_slab, &fs_handle->dpi_write_slab);
-	if (rc != -DER_SUCCESS)
-		D_GOTO(err_slab, rc);
+		rc = d_slab_register(&fs_handle->dpi_slab, &read_slab, eqt, &eqt->de_read_slab);
+		if (rc != -DER_SUCCESS)
+			D_GOTO(err_slab, rc);
 
-	rc = pthread_create(&fs_handle->dpi_thread, NULL, dfuse_progress_thread, fs_handle);
-	if (rc != 0)
-		D_GOTO(err_slab, rc = daos_errno2der(rc));
+		rc = d_slab_register(&fs_handle->dpi_slab, &write_slab, eqt, &eqt->de_write_slab);
+		if (rc != -DER_SUCCESS)
+			D_GOTO(err_slab, rc);
 
-	pthread_setname_np(fs_handle->dpi_thread, "dfuse_progress");
+		rc = pthread_create(&eqt->de_thread, NULL, dfuse_progress_thread, eqt);
+		if (rc != 0)
+			D_GOTO(err_threads, rc = daos_errno2der(rc));
```

Comment:
Since these threads are created in a loop, shouldn't other error handling above `D_GOTO(err_threads, rc)`?. For example, if any iteration after `i=0` fails